sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -251,44 +308,62 @@ class GpuDynamicPartitionDataWriter(
   /** Evaluates the `partitionPathExpression` above on a row of `partitionValues` and returns
    * the partition string.
    */
-  private lazy val getPartitionPath: InternalRow => String = {
+  protected lazy val getPartitionPath: InternalRow => String = {
     val proj = UnsafeProjection.create(Seq(partitionPathExpression), description.partitionColumns)
     row => proj(row).getString(0)
   }
 
+  /** Release resources of writer. */
+  private [rapids] def releaseWriter(writer: ColumnarOutputWriter): Unit = {
+    if (writer != null) {
+      writer.close()
+      statsTrackers.foreach(_.closeFile(writer.path()))
```

Comment:
nit: should we save the path before we close the writer? I know it works now, but using something after it is closed feels a little bad.

Reply:
Done